timemachine/fe/reweighting.py

```diff
@@ -0,0 +1,309 @@
+__all__ = [
+    "one_sided_exp",
+    "construct_endpoint_reweighting_estimator",
+    "construct_mixture_reweighting_estimator",
+    "interpret_as_mixture_potential",
+]
+
+from typing import Callable, Collection
+
+from jax import numpy as np
+from jax.scipy.special import logsumexp
+
+Samples = Collection
+Params = Collection
+Array = np.ndarray
+Energies = Array
+
+BatchedReducedPotentialFxn = Callable[[Samples, Params], Energies]
+
+
+def log_mean(log_values: Array) -> float:
+    """stable log(mean(values))
+
+    log(mean(values))
+    = log(sum(values / len(values)))
+    = logsumexp(log(values) - log(len(values))
+    """
+    return logsumexp(log_values - np.log(len(log_values)))
+
+
+def estimate_log_z_ratio(log_importance_weights: Array) -> float:
+    """stable log(mean(importance_weights))"""
+    return log_mean(log_importance_weights)
+
+
+def one_sided_exp(delta_us: Array) -> float:
+    """exponential averaging
+
+    References
+    ----------
+    [1] pymbar implementation
+        https://github.com/choderalab/pymbar/blob/15f932a271343e611ed4be2d468c77b1d11cf01f/pymbar/exp.py#L54
+    """
+    # delta_us = -log_importance_weights
+    # delta_f  = -log_z_ratio
```

Comment:
extraneous?

Reply:
Previous feedback was to document sign flips / logs when switching between energies etc. <-> log probabilities etc. -- will remove the extraneous comments

Reply:
Oops I see! Totally fine to leave it here then!